### PR TITLE
Nuka Cocktail Update

### PR DESCRIPTION
**Pull request by scarwolff**
## Description

Adds 21 new NUKA cocktails to compensate for blitzing tgheads drinks. Its a start. No wiki yet so recipes are here. They all do different fun and mechanically useful stuff. Experiment and see! If their effects are completely bugged and OP, compare it to a stimpack and get back to me.

/datum/chemical_reaction/nukacherry
	name = "Nuka Cherry"
	id = "nukacherry"
	results = list("nukacherry" = 2)
	required_reagents = list("nuka_cola" = 1, "cherryjelly" = 1)

/datum/chemical_reaction/nukagrape
	name = "Nuka Grape"
	id = "nukagrape"
	results = list("nukagrape" = 2)
	required_reagents = list("nuka_cola" = 1, "grapejuice" = 1)

/datum/chemical_reaction/nukadark
	name = "Nuka Dark"
	id = "nukadark"
	results = list("nukadark" = 2)
	required_reagents = list("nuka_cola" = 1, "buffalo" = 1)

/datum/chemical_reaction/nukaorange
	name = "Nuka Orange"
	id = "nukaorange"
	results = list("nukaorange" = 2)
	required_reagents = list("nuka_cola" = 1, "orangejuice" = 1)

/datum/chemical_reaction/nukaquartz
	name = "Nuka Quartz"
	id = "nukaquartz"
	results = list("nukaquartz" = 3)
	required_reagents = list("nuka_cola" = 1, "milk" = 1, "sugar" = 1)

/datum/chemical_reaction/nukavictory
	name = "Nuka Victory"
	id = "nukavictory"
	results = list("nukavictory" = 2)
	required_reagents = list("nuka_cola" = 1, "alliescocktail" = 1)

/datum/chemical_reaction/nukawild
	name = "Nuka Wild"
	id = "nukawild"
	results = list("nukawild" = 2)
	required_reagents = list("nuka_cola" = 1, "sunset" = 1)

/datum/chemical_reaction/nukaice
	name = "Iced Nuka"
	id = "nukaice"
	results = list("nukaice" = 2)
	required_reagents = list("nuka_cola" = 1, "ice" = 1)

/datum/chemical_reaction/nukaquantum
	name = "Nuka Quantum"
	id = "nukaquantum"
	results = list("nukaquantum" = 3)
	required_reagents = list("nuka_cola" = 1, "cleaner" = 1, "bluespace" = 1)

//nuka mixes

/datum/chemical_reaction/nukanew
	name = "Newka Cola"
	id = "nukanew"
	results = list("nukanew" = 3)
	required_reagents = list("nukacherry" = 1, "nukawild" = 1, "ice" = 1)

/datum/chemical_reaction/nukaberry
	name = "Nuka Berry"
	id = "nukaberry"
	results = list("nukaberry" = 3)
	required_reagents = list("nukacherry" = 1, "nukagrape" = 1, "ice" = 1)

/datum/chemical_reaction/nukabomb
	name = "Nuka Bombdrop"
	id = "nukabomb"
	results = list("nukabomb" = 5)
	required_reagents = list("nukadark" = 1, "rum" = 1, "vodka" = 1, "whiskey" = 1, "sugar" = 1)

/datum/chemical_reaction/nukacide
	name = "Nukacide"
	id = "nukacide"
	results = list("nukacide" = 3)
	required_reagents = list("nukabomb" = 1, "nukaquantum" = 1, "ice" = 1)

/datum/chemical_reaction/nukacooler
	name = "Nuka Cooler"
	id = "nukacooler"
	results = list("nukacooler" = 3)
	required_reagents = list("nukaice" = 1, "ice" = 1, "water" = 1)

/datum/chemical_reaction/nukafancy
	name = "Nuka Fancy"
	id = "nukafancy"
	results = list("nukafancy" = 3)
	required_reagents = list("nuka_cola" = 1, "martini" = 1, "ice" = 1)

/datum/chemical_reaction/nukafree
	name = "Diet Nuka"
	id = "nukafree"
	results = list("nukafree" = 3)
	required_reagents = list("nuka_cola" = 1, "sodiumchloride" = 1, "water" = 1)

/datum/chemical_reaction/nukafrutti
	name = "Nuka Frutti"
	id = "nukafrutti"
	results = list("nukafrutti" = 3)
	required_reagents = list("nukaberry" = 1, "nukaorange" = 1, "ice" = 1)

/datum/chemical_reaction/nukalove
	name = "Nuka Love"
	id = "nukalove"
	results = list("nukalove" = 3)
	required_reagents = list("nuka_cola" = 1, "between_the_sheets" = 1, "ice" = 1)

/datum/chemical_reaction/nukapunch
	name = "Nuka Punch"
	id = "nukapunch"
	results = list("nukapunch" = 3)
	required_reagents = list("nukadark" = 1, "deathroach" = 1, "pungajuice" = 1)

/datum/chemical_reaction/nukasunrise
	name = "Nuka Sunrise"
	id = "nukasunrise"
	results = list("nukasunrise" = 5)
	required_reagents = list("nukawild" = 2, "orangejuice" = 2, "grenadine" = 1)

/datum/chemical_reaction/nukaxtreme
	name = "Nuka X-Treme"
	id = "nukaxtreme"
	results = list("nukaxtreme" = 1)
	required_reagents = list("nukacide" = 10, "nukaquantum" = 10, "ice" = 10)

## How Has This Been Tested?

Actually compiles, also fixed the prior bugs.
## Screenshots (if appropriate):
image

## Changelog (necessary)
:cl:
add: Added 21 new nuka cola style cocktails, all with individual perks and recipes.
tweak: fixed the last commit
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/recipes/drinks_recipes.dm [hunk 1]
```diff
@@ -433,4 +433,136 @@
...
+	name = "Nuka Sunrise"
+	id = "nukasunrise"
+	results = list("nukasunrise" = 5)
+	required_reagents = list("nukawild" = 2, "orangejuice" = 2, "grenadine" = 1)
+
+/datum/chemical_reaction/nukaxtreme
+	name = "Nuka X-Treme"
+	id = "nukaxtreme"
+	results = list("nukaxtreme" = 1)
+	required_reagents = list("nukacide" = 10, "nukaquantum" = 10, "ice" = 10)
```
Comment (Laus4Deus) [inline code/modules/food_and_drinks/recipes/drinks_recipes.dm:568]: Are you sure 10u of each to yield 1u of X-treme?

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 2]
```diff
@@ -1299,3 +1299,317 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+	id = "nukabomb"
+	description = "More spirit than Nuka at this Rate."
+	color = "#FAEBD7"
+	boozepwr = 200
+	taste_description = "pure alcohol"
+	glass_icon_state = "nukabombglass"
+	glass_name = "Nuka Bombdrop"
+	glass_desc = "More spirit than Nuka at this Rate."
+
+/datum/reagent/consumable/ethanol/nukabomb/on_mob_life(mob/living/carbon/M)
```
Comment (Laus4Deus) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:1364]: 🤣

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 3]
```diff
@@ -1299,3 +1299,317 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+	description = "The drink of a goddamn madman, say your sorrows when you drink this."
+	color = "#000000"
+	boozepwr = 300
+	taste_description = "nuclear annihilation"
+	glass_icon_state = "nukacideglass"
+	glass_name = "Nukacide"
+	glass_desc = "The drink of a goddamn madman, say your sorrows when you drink this."
+
+/datum/reagent/consumable/ethanol/nukacide/on_mob_life(mob/living/carbon/M)
+	M.vomit(100)
```
Comment (Laus4Deus) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:1394]: you should set vomit to 90, so it's not a 100% guarantee vomit

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 4]
```diff
@@ -1299,3 +1299,317 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+	description = "The Refined mans Soda, Fit for soda royalty."
+	color = "#11111E"
+	boozepwr = 30
+	taste_description = "refined soda"
+	glass_icon_state = "nukafancyglass"
+	glass_name = "Nuka Fancy"
+	glass_desc = "The Refined mans Soda, Fit for soda royalty."
+
+/datum/reagent/consumable/ethanol/nukafancy/on_mob_life(mob/living/carbon/M)
+	var/high_message = pick("<br>Maybe I too need some Slaves?</b>","<br>Mutfruit for All!</b>","<br>Time to Glorify my Wasteland Castle!</b>","<brNuked, not stirred.</b>")
```
Comment (Laus4Deus) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:1416]: typo
`<brNuked`

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 5]
```diff
@@ -1299,3 +1299,317 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+	name = "Nuka Quantum"
+	id = "nukaquantum"
+	description = "An extremely blue and glowing combination of Nuka-Cola and (REDACTED)."
+	color = "#6AFFFF"
+	boozepwr = 10
+	taste_description = "the eighteenth flavour"
+	glass_icon_state = "nukaquantumglass"
+	glass_name = "Nuka Quantum"
+	glass_desc = "An extremely blue and glowing combination of Nuka-Cola and (REDACTED)"
+	var/datum/brain_trauma/special/psychotic_brawling/bath_salts/rage
```
Comment (Laus4Deus) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:1510]: Woah, so it fucks you up 4 the rest of the round?

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 6]
```diff
@@ -1299,3 +1299,317 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+	M.adjust_bodytemperature(-5 * TEMPERATURE_DAMAGE_COEFFICIENT, BODYTEMP_NORMAL)
+	M.adjustBrainLoss(4,0)
+	M.adjustToxLoss(4, 0)
+	M.AdjustStun(-30, 0)
+	M.AdjustKnockdown(-30, 0)
+	M.AdjustUnconscious(-30, 0)
+	M.adjustStaminaLoss(-5, 0)
+	M.hallucination += 20
+	M.Jitter(2)
+	M.add_trait(TRAIT_IRONFIST, id)
```
Comment (Laus4Deus) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:1577]: oh god

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 7]
```diff
@@ -1299,3 +1299,317 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+	..()
+	. = 1
+
+/datum/reagent/consumable/ethanol/nukaxtreme/on_mob_life(mob/living/carbon/M)
+	if(M.hud_used)
+		if(current_cycle >= 5 && current_cycle % 3 == 0)
+			var/list/screens = list(M.hud_used.plane_masters["[FLOOR_PLANE]"], M.hud_used.plane_masters["[GAME_PLANE]"])
+			var/matrix/skew = matrix()
+			var/intensity = 8
+			skew.set_skew(rand(-intensity,intensity), rand(-intensity,intensity))
```
Comment (Laus4Deus) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:1592]: ¿¡SKEWIUM!?

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 8]
```diff
@@ -835,3 +834,240 @@
...
+	id = "nukaberry"
+	description = "Nuka-Cola with a Berry Aftertaste."
+	color = "#753453"
+	taste_description = "berry Nuka-Cola"
+	glass_icon_state = "nukaberryglass"
+	glass_name = "Nuka Berry"
+	glass_desc = "Nuka-Cola with a Berry Aftertaste."
+
+/datum/reagent/consumable/nukaberry/on_mob_life(mob/living/carbon/M)
+	M.adjustFireLoss(-2*REM, 0)
```
Comment (Laus4Deus) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:998]: this one might be too stronk

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 9]
```diff
@@ -835,3 +834,240 @@
...
+	color = "#663300"
+	taste_description = "watered down Nuka"
+	glass_icon_state = "nukafreeglass"
+	glass_name = "Diet Nuka"
+	glass_desc = "Nuka-Cola with no added sugars!"
+
+/datum/reagent/consumable/nukafree/on_mob_life(mob/living/carbon/M)
+	if(M.nutrition <= NUTRITION_LEVEL_STARVING)
+		M.adjustToxLoss(1*REM, 0)
+	M.nutrition = max(M.nutrition - 3, 0)
```
Comment (Laus4Deus) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:1041]: Diet!

File: code/modules/food_and_drinks/recipes/drinks_recipes.dm (hunk 1 above)
Comment (scarwolff) [reply code/modules/food_and_drinks/recipes/drinks_recipes.dm:568]: this shit kind wacky man, dont want it to be easy to make

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm (hunk 3 above)
Comment (scarwolff) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:1394]: eh, its a nukacide shot, your lucky it dosent outright kill you

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm (hunk 5 above)
Comment (scarwolff) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:1510]: nah just for as long as its in your system, look at the on mob delete for brain trauma

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm (hunk 7 above)
Comment (scarwolff) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:1592]: EXTREME

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 8 above)
Comment (scarwolff) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:998]: i mean its still weaker than a stimpack

**Pull request merged**